### ISSUE TITLE
Add logs and update the docs to inform that FIPS is intentionally disabled for OpenSSL1.1.1 and OpenSSL 1.0.2 series

### ISSUE THREAD

**Issue by favipcj** [zhaozg/lua-openssl#298]
**Is your feature request related to a problem? Please describe.**

When using this library in environments that have OpenSSL1.1.1 or OpenSSL1.0.2 and FIPS is enabled. It is not possible to know that FIPS is intentionally been disabled for those OpenSSL series, as there is no documentation or logs pointing that out.

Only after reviewing the code, you can find that FIPS is being disabled when initializing this library, which is happening [here](https://github.com/zhaozg/lua-openssl/blob/master/src/openssl.c#L345). This is needed as in most projects there are many libraries playing together and it's difficult to review the code of each of them to find out what they are doing.

**NOTE:** FIPS is not correctly disabled in environments that have OpenSSL 1.1.1 installed, so this is causing nginx to fail when executing openssl handshakes after importing this library. Related to https://github.com/openssl/openssl/issues/15144
```
SSL_do_handshake() failed (SSL: error:060800C8:digital envelope routines:EVP_DigestInit_ex:disabled for FIPS error:060800C8:digital envelope routines:EVP_DigestInit_ex:disabled for FIPS) while SSL handshaking
```

**Describe the solution you'd like**
There are 2 actions that I'm suggesting:
* Add WARNING logs to inform when FIPS is intentionally disabled and the reason why. i.e: `WARNING: FIPS is being disabled because OpenSSL 1.X is being used, which is not supported by this library....`
   - I'm not so sure how to add logs, but if you can guide me with some examples, I can open a PR for this too.

* Add something like the following to the documentation or to the project README, so developers can be aware of this before using this library.
   - **WARNING**: This library does not support FIPS for OpenSSL versions minor to OpenSSL3. (I can open a PR to add this)

**Describe alternatives you've considered**
Other suggestions that need more work and analysis:
 - support FIPS for OpenSSL 1.1.1 series, as OpenSSL3 is still in the process of certification.
 - if FIPS is not going to be supported for OpenSSL 1.1.1 series, it should be disabled as expected. This is another issue as this is not working in environments that have OpenSSL 1.1.1 installed. FIPS was disabled only after executing this `require("resty.openssl").set_fips_mode(false)`.

**Additional context**

This was discovered using the following environment:
- docker image: UBI8 (Rhel8) or OL8
- OpenSSL 1.1.1k (or 1.1.1g)
- nginx version: 1.21.6
- OpenResty: 1.21.4.1
- lua-openssl: 0.8.5-1

**Comment by zhaozg** [comment]
Thank your for clean and clear description of the problem.

1. The https://github.com/zhaozg/lua-openssl/blob/4fa9f76495bc0dbd649b0b7f63fa422acb977312/src/openssl.c#L350 line should be remove, this line is not valuable.
2. You can call `require('openssl').FIPS_mode(true)` to enable FIPS mode manually. and call `return('openssl').FIPS_mode()` to get openssl support FIPS mode or not.

**Comment by zhaozg** [comment]
closed in https://github.com/zhaozg/lua-openssl/commit/4395ef06ca135fb0cd9639cfa4bbaa0116d3d581

**Comment by favipcj** [comment]
oh nice!, thanks for fixing this. I'll test this out on Monday.
